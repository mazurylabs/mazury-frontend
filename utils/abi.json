{
  "status": "1",
  "message": "OK",
  "result": [
    {
      "inputs": [
        {
          "internalType": "string",
          "name": "_contractBaseURI",
          "type": "string"
        }
      ],
      "stateMutability": "nonpayable",
      "type": "constructor"
    },
    {
      "anonymous": false,
      "inputs": [
        {
          "indexed": true,
          "internalType": "address",
          "name": "owner",
          "type": "address"
        },
        {
          "indexed": true,
          "internalType": "address",
          "name": "approved",
          "type": "address"
        },
        {
          "indexed": true,
          "internalType": "uint256",
          "name": "tokenId",
          "type": "uint256"
        }
      ],
      "name": "Approval",
      "type": "event"
    },
    {
      "anonymous": false,
      "inputs": [
        {
          "indexed": true,
          "internalType": "address",
          "name": "owner",
          "type": "address"
        },
        {
          "indexed": true,
          "internalType": "address",
          "name": "operator",
          "type": "address"
        },
        {
          "indexed": false,
          "internalType": "bool",
          "name": "approved",
          "type": "bool"
        }
      ],
      "name": "ApprovalForAll",
      "type": "event"
    },
    {
      "anonymous": false,
      "inputs": [
        {
          "indexed": true,
          "internalType": "address",
          "name": "_from",
          "type": "address"
        },
        {
          "indexed": true,
          "internalType": "string",
          "name": "_slug",
          "type": "string"
        },
        {
          "indexed": true,
          "internalType": "uint256",
          "name": "_badgeIndex",
          "type": "uint256"
        }
      ],
      "name": "Burn",
      "type": "event"
    },
    {
      "anonymous": false,
      "inputs": [
        {
          "indexed": true,
          "internalType": "address",
          "name": "_receiver",
          "type": "address"
        },
        {
          "indexed": true,
          "internalType": "string",
          "name": "_slug",
          "type": "string"
        },
        {
          "indexed": true,
          "internalType": "uint256",
          "name": "_badgeIndex",
          "type": "uint256"
        }
      ],
      "name": "Claim",
      "type": "event"
    },
    {
      "anonymous": false,
      "inputs": [
        {
          "indexed": true,
          "internalType": "address",
          "name": "previousOwner",
          "type": "address"
        },
        {
          "indexed": true,
          "internalType": "address",
          "name": "newOwner",
          "type": "address"
        }
      ],
      "name": "OwnershipTransferred",
      "type": "event"
    },
    {
      "anonymous": false,
      "inputs": [
        {
          "indexed": true,
          "internalType": "address",
          "name": "from",
          "type": "address"
        },
        {
          "indexed": true,
          "internalType": "address",
          "name": "to",
          "type": "address"
        },
        {
          "indexed": true,
          "internalType": "uint256",
          "name": "tokenId",
          "type": "uint256"
        }
      ],
      "name": "Transfer",
      "type": "event"
    },
    {
      "inputs": [
        { "internalType": "address", "name": "to", "type": "address" },
        { "internalType": "uint256", "name": "tokenId", "type": "uint256" }
      ],
      "name": "approve",
      "outputs": [],
      "stateMutability": "nonpayable",
      "type": "function"
    },
    {
      "inputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }],
      "name": "badgeTypes",
      "outputs": [
        { "internalType": "string", "name": "slug", "type": "string" },
        { "internalType": "uint256", "name": "badgeMinted", "type": "uint256" },
        { "internalType": "uint256", "name": "id", "type": "uint256" }
      ],
      "stateMutability": "view",
      "type": "function"
    },
    {
      "inputs": [
        { "internalType": "address", "name": "owner", "type": "address" }
      ],
      "name": "balanceOf",
      "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }],
      "stateMutability": "view",
      "type": "function"
    },
    {
      "inputs": [
        { "internalType": "address", "name": "", "type": "address" },
        { "internalType": "uint256", "name": "", "type": "uint256" }
      ],
      "name": "claimed",
      "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }],
      "stateMutability": "view",
      "type": "function"
    },
    {
      "inputs": [
        { "internalType": "uint256", "name": "id", "type": "uint256" },
        { "internalType": "string", "name": "slug", "type": "string" }
      ],
      "name": "createBadgeType",
      "outputs": [],
      "stateMutability": "nonpayable",
      "type": "function"
    },
    {
      "inputs": [
        { "internalType": "uint256", "name": "tokenId", "type": "uint256" }
      ],
      "name": "getApproved",
      "outputs": [{ "internalType": "address", "name": "", "type": "address" }],
      "stateMutability": "view",
      "type": "function"
    },
    {
      "inputs": [
        { "internalType": "address", "name": "owner", "type": "address" },
        { "internalType": "address", "name": "operator", "type": "address" }
      ],
      "name": "isApprovedForAll",
      "outputs": [{ "internalType": "bool", "name": "", "type": "bool" }],
      "stateMutability": "view",
      "type": "function"
    },
    {
      "inputs": [
        { "internalType": "address", "name": "to", "type": "address" },
        { "internalType": "uint256", "name": "id", "type": "uint256" }
      ],
      "name": "issueToken",
      "outputs": [],
      "stateMutability": "nonpayable",
      "type": "function"
    },
    {
      "inputs": [],
      "name": "name",
      "outputs": [{ "internalType": "string", "name": "", "type": "string" }],
      "stateMutability": "view",
      "type": "function"
    },
    {
      "inputs": [],
      "name": "owner",
      "outputs": [{ "internalType": "address", "name": "", "type": "address" }],
      "stateMutability": "view",
      "type": "function"
    },
    {
      "inputs": [
        { "internalType": "uint256", "name": "tokenId", "type": "uint256" }
      ],
      "name": "ownerOf",
      "outputs": [{ "internalType": "address", "name": "", "type": "address" }],
      "stateMutability": "view",
      "type": "function"
    },
    {
      "inputs": [],
      "name": "renounceOwnership",
      "outputs": [],
      "stateMutability": "nonpayable",
      "type": "function"
    },
    {
      "inputs": [
        { "internalType": "address", "name": "from", "type": "address" },
        { "internalType": "address", "name": "to", "type": "address" },
        { "internalType": "uint256", "name": "tokenId", "type": "uint256" }
      ],
      "name": "safeTransferFrom",
      "outputs": [],
      "stateMutability": "nonpayable",
      "type": "function"
    },
    {
      "inputs": [
        { "internalType": "address", "name": "from", "type": "address" },
        { "internalType": "address", "name": "to", "type": "address" },
        { "internalType": "uint256", "name": "tokenId", "type": "uint256" },
        { "internalType": "bytes", "name": "data", "type": "bytes" }
      ],
      "name": "safeTransferFrom",
      "outputs": [],
      "stateMutability": "nonpayable",
      "type": "function"
    },
    {
      "inputs": [
        { "internalType": "bool", "name": "_allowsTransfers", "type": "bool" }
      ],
      "name": "setAllowsTransfers",
      "outputs": [],
      "stateMutability": "nonpayable",
      "type": "function"
    },
    {
      "inputs": [
        { "internalType": "address", "name": "operator", "type": "address" },
        { "internalType": "bool", "name": "approved", "type": "bool" }
      ],
      "name": "setApprovalForAll",
      "outputs": [],
      "stateMutability": "nonpayable",
      "type": "function"
    },
    {
      "inputs": [
        { "internalType": "bool", "name": "_burnState", "type": "bool" }
      ],
      "name": "setBurnState",
      "outputs": [],
      "stateMutability": "nonpayable",
      "type": "function"
    },
    {
      "inputs": [
        {
          "internalType": "string",
          "name": "_contractBaseURI",
          "type": "string"
        }
      ],
      "name": "setContractBaseURI",
      "outputs": [],
      "stateMutability": "nonpayable",
      "type": "function"
    },
    {
      "inputs": [
        { "internalType": "bytes4", "name": "interfaceId", "type": "bytes4" }
      ],
      "name": "supportsInterface",
      "outputs": [{ "internalType": "bool", "name": "", "type": "bool" }],
      "stateMutability": "view",
      "type": "function"
    },
    {
      "inputs": [],
      "name": "symbol",
      "outputs": [{ "internalType": "string", "name": "", "type": "string" }],
      "stateMutability": "view",
      "type": "function"
    },
    {
      "inputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }],
      "name": "tokenIdToBadgeTypeId",
      "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }],
      "stateMutability": "view",
      "type": "function"
    },
    {
      "inputs": [
        { "internalType": "uint256", "name": "tokenId", "type": "uint256" }
      ],
      "name": "tokenURI",
      "outputs": [{ "internalType": "string", "name": "", "type": "string" }],
      "stateMutability": "view",
      "type": "function"
    },
    {
      "inputs": [
        { "internalType": "address", "name": "from", "type": "address" },
        { "internalType": "address", "name": "to", "type": "address" },
        { "internalType": "uint256", "name": "tokenId", "type": "uint256" }
      ],
      "name": "transferFrom",
      "outputs": [],
      "stateMutability": "nonpayable",
      "type": "function"
    },
    {
      "inputs": [
        { "internalType": "address", "name": "newOwner", "type": "address" }
      ],
      "name": "transferOwnership",
      "outputs": [],
      "stateMutability": "nonpayable",
      "type": "function"
    },
    {
      "inputs": [
        { "internalType": "address", "name": "_admin", "type": "address" },
        { "internalType": "bool", "name": "isAdmin", "type": "bool" }
      ],
      "name": "updateAdmin",
      "outputs": [],
      "stateMutability": "nonpayable",
      "type": "function"
    }
  ]
}
