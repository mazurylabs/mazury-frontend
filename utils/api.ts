import { OnboardingFormDataType } from 'contexts';
import { Activity, APIResponse, ListResponse, Referral } from 'types';
import { api } from '.';

export const getProfile = async (address: string) => {
  try {
    const res = await api.get(`/profiles/${address}`);
    return {
      data: res.data,
      error: null,
    };
  } catch (error) {
    return {
      data: null,
      error,
    };
  }
};

export const getMessageToBeSigned: (
  address: string
) => Promise<APIResponse<string | null>> = async (address) => {
  try {
    const res = await api.get(`/auth/message?address=${address}`);
    return {
      data: res.data,
      error: null,
    };
  } catch (error) {
    return {
      data: null,
      error,
    };
  }
};

export const updateProfile: (
  address: string,
  signature: string,
  data: OnboardingFormDataType,
  avatarFile?: File | null
) => Promise<APIResponse> = async (address, signature, data, avatarFile) => {
  try {
    const formData = new FormData();
    for (let key in data) {
      if (key === 'avatar') {
        continue;
      }
      // @ts-ignore
      formData.append(key, data[key]);
    }
    if (avatarFile) {
      formData.append('avatar', avatarFile, avatarFile.name);
    }
    const res = await api.patch(`/profiles/${address}/`, formData, {
      headers: {
        'ETH-AUTH': signature,
      },
    });
    return {
      data,
      error: null,
    };
  } catch (err) {
    return {
      data: null,
      error: err,
    };
  }
};

export const getActvity: (
  address?: string
) => Promise<APIResponse<ListResponse<Activity>>> = async (address) => {
  try {
    const res = await api.get(`/activity?user=${address}`);
    return {
      data: res.data,
      error: null,
    };
  } catch (error) {
    return {
      data: null,
      error,
    };
  }
};

export const createReferral: (
  authorAddress: string,
  receiverAddress: string,
  content: string,
  skills: string[],
  authorSignature: string
) => Promise<APIResponse<Referral>> = async (
  authorAddress,
  receiverAddress,
  content,
  skills,
  authorSignature
) => {
  try {
    const res = await api.post(
      '/referrals/',
      {
        receiver: {
          eth_address: receiverAddress,
        },
        author: {
          eth_address: authorAddress,
        },
        content,
        skills,
      },
      {
        headers: {
          'ETH-AUTH': authorSignature,
        },
      }
    );
    return {
      data: res.data,
      error: null,
    };
  } catch (error) {
    return {
      data: null,
      error,
    };
  }
};
