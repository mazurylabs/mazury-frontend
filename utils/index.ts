import { API_URL } from '@/config';
// import { axios } from '@/lib/axios';
import axios from 'axios';
import { ParsedUrlQuery } from 'querystring';
import type {
  Badge,
  ColorName,
  Profile,
  ProfileSection,
  Referral,
  Skill,
} from 'types';
import { theme } from '../tailwind.config';

export const getTruncatedAddress = (
  address: string | undefined | null,
  length: number = 4
): string => {
  if (!address) {
    return '';
  }

  return `${address.slice(0, length + 2)}...${address.slice(
    address.length - length
  )}`;
};

export const goToLink = (link: string) => {
  window.open(link, '_blank');
};

function capitalize(word: string) {
  return word.charAt(0).toUpperCase() + word.substring(1);
}

export function toCapitalizedWord(word: string) {
  var words = word.match(/[A-Za-z][a-z]*/g) || [];
  return words.map(capitalize).join(' ');
}

export const colors = theme.extend.colors;

type MonthDigit = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;

export const getMonthName = (month: MonthDigit) => {
  const months = [
    'January',
    'February',
    'March',
    'April',
    'May',
    'June',
    'July',
    'August',
    'September',
    'October',
    'November',
    'December',
  ];
  return months[month - 1];
};

export const getMonthAndYear = (date: Date) => {
  const month = date.getMonth() + 1;
  const year = date.getFullYear();
  return `${getMonthName(month as MonthDigit).slice(0, 3)} ${year}`;
};

export const skillsList: Skill[] = [
  'frontendDev',
  'backendDev',
  'smartContractDev',
  'protocolDev',
  'design',
  'growthMarketing',
  'writing',
  'productManagement',
  'projectManagement',
  'dataScience',
  'art',
  'defiDegen',
  'nftDegen',
  'teaching',
  'memes',
  'community',
];

export const toCamelCase = (text: string) => {
  return text
    .replace(/(?:^\w|[A-Z]|\b\w)/g, (leftTrim, index) =>
      index === 0 ? leftTrim.toLowerCase() : leftTrim.toUpperCase()
    )
    .replace(/\s+/g, '');
};

// Scans an array of referrals and finds out if the user has already referred someone
export const hasAlreadyReferredReceiver = (
  referrals: Referral[],
  receiver: string,
  author: string
) => {
  return referrals.find((referral) => {
    return (
      referral.receiver.eth_address === receiver &&
      referral.author.eth_address === author
    );
  });
};

export const getTwitterConnectionPopupLink = (ethAddress: string) => {
  return `https://twitter.com/intent/tweet?text=I'm%20verifying%20myself%20for%20%40mazuryxyz%20%F0%9F%8C%8A%0a%0ahttps://mzry.me/${ethAddress}`;
};

/**
 * Returns '-' if the value is null or undefined, otherwise returns the value. Useful because 0 || '-' returns '-'
 */
export const getMetricDisplayValue = (
  value: number | null | undefined,
  placeholder: string = '-'
) => {
  return value === null || value === undefined ? placeholder : value;
};

const detectIfEthAddress = (str: string) => {
  return /^(0x)?[0-9a-f]{40}$/i.test(str);
};

export const returnTruncatedIfEthAddress = (str: string) => {
  if (detectIfEthAddress(str)) {
    return getTruncatedAddress(str);
  }
  return str;
};

export const commify = (value: number) => {
  let numberString: string | string[] = String(value);

  numberString = numberString.split('').reverse();

  numberString = numberString.reduce((prev, next, index) => {
    let shouldComma = (index + 1) % 3 === 0 && index + 1 < numberString.length;

    let updatedValue = `${prev}${next}`;

    if (shouldComma) {
      updatedValue = `${updatedValue},`;
    }

    return updatedValue;
  }, '');

  numberString = numberString.split('').reverse().join('');

  return numberString;
};

export const getSkillSlugsFromReferral = (referral: Referral) => {
  const { skills } = referral;
  return skills?.map((skill) => skill.slug);
};

const getOffset = (page: number) => {
  return (page - 1) * 20;
};

export const getOffsetArray = (page: number) => {
  const offsets = [];
  for (let i = 1; i <= page; i++) {
    offsets.push(getOffset(i));
  }
  return offsets;
};

export const getSkillsFromProfile = (profile: Partial<Profile>) => {
  const skills = [];
  for (let skill of skillsList) {
    if (profile[skill]) {
      skills.push(skill);
    }
  }
  return skills;
};

export const sectionToColor: { [key in ProfileSection]: ColorName } = {
  Activity: 'indigo',
  Credentials: 'fuchsia',
  Referrals: 'emerald',
  Writing: 'amber',
  DAOs: 'purple',
};

export const truncateString = (str: string, maxLength: number = 50) => {
  return str.length > maxLength ? `${str.slice(0, maxLength)}...` : str;
};

export const trayAnimation = {
  initial: { y: '100%' },
  animate: {
    y: 0,
    transition: { duration: 0.4, ease: [0.36, 0.66, 0.04, 1] },
  },
  exit: {
    y: '100%',
    transition: { duration: 0.3, ease: [0.36, 0.66, 0.04, 1] },
  },
};

export const fadeAnimation = {
  initial: { opacity: 0 },
  animate: {
    opacity: 1,
  },
  exit: {
    opacity: 0,
  },
};

export const getRoles = (data: Profile, query: ParsedUrlQuery) => {
  let keys = Object.entries(data);

  let roles = keys.map(([key, value]) => {
    if (key.startsWith('role') && value) {
      let role = key.split('_')[1];

      return {
        title: role,
        hightlighted: (query?.role as string)?.includes(role),
      };
    }
  });

  const validRoles = roles.filter((role) => Boolean(role)).slice(0, 2);
  let remainder = validRoles.length - 2;

  return { roles: validRoles, remainder: remainder < 1 ? 0 : remainder };
};

export const getHighlightedBadges = (data: Badge[], query: ParsedUrlQuery) => {
  let badges = data.map((badge) => {
    const title = badge.badge_type.title;
    const hightlighted =
      (query?.badges as string)?.slice(1)?.split('_')?.join(' ') ===
      title.toLowerCase();

    return {
      title,
      hightlighted,
    };
  });

  const validBadges = badges.slice(0, 2);
  let remainder = validBadges.length - 2;

  return { badges: validBadges, remainder: remainder < 1 ? 0 : remainder };
};

const wagmiSetupKeys = [
  'wagmi.wallet',
  'walletconnect',
  '-walletlink:https://www.walletlink.org:session:id',
  '-walletlink:https://www.walletlink.org:session:secret',
  '-walletlink:https://www.walletlink.org:session:linked',
  '-walletlink:https://www.walletlink.org:IsStandaloneSigning',
];

export const clearWagmiStorage = () => {
  wagmiSetupKeys.forEach((key) => localStorage.removeItem(key));
};
