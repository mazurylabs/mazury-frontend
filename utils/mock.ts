import { Profile } from 'types';

export const mockProfile: Profile = {
  id: 'fc4d4d8a-f46e-4500-9361-7c8bb3737c05',
  top_badges: [
    {
      id: '976b84d5-d2ed-458b-982c-9c8a25883825',
      owner: {
        eth_address: '0x7f416e7a586d45ed2A4e77877E1d253f48105491',
        ens_name: '',
        username: '0x7f416e7a586d45ed2A4e77877E1d253f48105491',
        avatar:
          'https://mazurylabs-staging.s3.eu-central-1.amazonaws.com/gradient4.jpg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAYEYJ2JINA57WRDY7%2F20220419%2Feu-central-1%2Fs3%2Faws4_request&X-Amz-Date=20220419T021718Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Signature=1b2bc3606f3ade788e6759ac305f2b6ccaf0afb8abce0bf4505021b9588165cb',
      },
      hidden: false,
      highlighted: false,
      badge_type: {
        id: '0046d0cf-2a66-4235-a8f3-062a0514253a',
        created_at: '2021-11-07T21:03:03.245105Z',
        updated_at: '2022-04-15T15:17:29.968195Z',
        image:
          'https://mazurylabs-staging.s3.eu-central-1.amazonaws.com/badges/uniswap.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAYEYJ2JINA57WRDY7%2F20220419%2Feu-central-1%2Fs3%2Faws4_request&X-Amz-Date=20220419T021718Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Signature=bf19284e8bf4f60b154cd434bbf5e160ebff56b85338a6173dd46b4a48be4ece',
        title: 'Uniswap proposal author',
        slug: 'uniswap_author',
        description: 'You published a proposal for uniswap',
        issuer: { name: 'mazury' },
        poap_badge_extra_data: null,
        video: '',
        tags: [{ level: '1', name: '' }],
      },
      created_at: '2021-11-28T21:16:23.817772Z',
      updated_at: '2021-11-28T21:16:23.817784Z',
      minted: false,
      minted_at: '',
    },
  ],
  referred_by: [],
  created_at: '2021-11-28T21:16:23.809084Z',
  updated_at: '2021-11-28T21:16:23.809098Z',
  avatar:
    'https://mazurylabs-staging.s3.eu-central-1.amazonaws.com/avatars/0xF417ACe7b13c0ef4fcb5548390a450A4B75D3eB3/IMG_2948.jpeg?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAYEYJ2JINA57WRDY7%2F20220421%2Feu-central-1%2Fs3%2Faws4_request&X-Amz-Date=20220421T213828Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Signature=a55b895d8c4d6f4e1209f46217d18d6587fbfdba7b7a77e6441233be2bb1a7ad',
  eth_address: '0x7f416e7a586d45ed2A4e77877E1d253f48105491',
  bio: '',
  username: 'test.eth',
  ens_name: '',
  claimed: false,
  signing_nonce: 1,
  verified: true,
  github: '',
  twitter: '',
  github_last_checked: null,
  open_to_opportunities: false,
  working_remotely: false,
  role_developer: false,
  role_designer: false,
  role_trader: false,
  role_creator: false,
  role_researcher: false,
  role_investor: false,
  role_community_manager: false,
  onboarded: false,
  email: '',
  website: '',
  full_name: '',
  is_recruiter: false,
  how_did_you_find_us: '',
  linkedIn: '',
  location: '',
};
