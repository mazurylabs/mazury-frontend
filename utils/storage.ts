import jwt_decode from 'jwt-decode';
import { APP_STORAGE_PREFIX } from '@/config';

const storage = {
  getToken: (key: string) => {
    return JSON.parse(
      window.localStorage.getItem(APP_STORAGE_PREFIX + key) as string
    );
  },
  setToken: (token: string, key: string) => {
    window.localStorage.setItem(
      APP_STORAGE_PREFIX + key,
      JSON.stringify(token)
    );
  },
  clearToken: (key: string) => {
    window.localStorage.removeItem(APP_STORAGE_PREFIX + key);
  },
  isTokenExpired: (token?: string) => {
    if (!token) return false;

    const decodedToken = jwt_decode(token) as { exp: number };

    return decodedToken?.exp < Date.now() / 1000;
  },
};

export default storage;
