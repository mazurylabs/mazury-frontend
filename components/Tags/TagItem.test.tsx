import { fireEvent, render, screen } from '@testing-library/react';
import { colors } from 'utils';
import { TagItem } from '.';

describe('TagItem', () => {
  test('renders correctly and fires onRemove as intended', () => {
    const onRemove = jest.fn();
    render(
      <TagItem
        color={colors.indigo}
        label="Frontend engineer"
        showRemove
        onRemove={onRemove}
        value="frontend"
      />
    );
    const container = screen.getByRole('tag');
    expect(container).toBeInTheDocument();
    expect(container).toHaveTextContent('Frontend engineer');
    expect(container).toMatchSnapshot();
    const removeButton = screen.getByRole('button');
    fireEvent.click(removeButton);
    expect(onRemove).toHaveBeenCalledTimes(1);
  });
});
