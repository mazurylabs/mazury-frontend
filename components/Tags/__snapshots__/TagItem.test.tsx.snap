// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagItem renders correctly and fires onRemove as intended 1`] = `
<div
  class="flex w-fit items-center gap-1 rounded px-2 py-1 text-xs font-bold undefined"
  role="tag"
  style="background-color: rgb(238, 242, 255); color: rgb(49, 46, 129);"
>
  <span
    class="cursor-default"
  >
    Frontend engineer
  </span>
  <button
    class="flex items-center hover:cursor-pointer"
  >
    <svg
      aria-label="Remove Frontend engineer"
      fill="none"
      height="16px"
      role="img"
      viewBox="0 0 16 16"
      width="16px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 4L4 12"
        stroke="#312e81"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.33"
      />
      <path
        d="M4 4L12 12"
        stroke="#312e81"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.33"
      />
    </svg>
  </button>
</div>
`;
